usaspending_api/download/management/commands/resolve_frozen_downloads.py

```diff
@@ -0,0 +1,102 @@
+import dateutil
+import logging
+import pytz
+import time
+
+from django.core.management.base import BaseCommand
+from django.db.models import Q
+
+from usaspending_api.download.lookups import JOB_STATUS_DICT
+from usaspending_api.download.models import DownloadJob
+from usaspending_api.download.v2.download_admin import DownloadAdministrator
+
+logger = logging.getLogger("console")
+
+
+class Command(BaseCommand):
+    help = " ".join(
+        [
+            "Restart all 'frozen' download jobs which meet the provided filters",
+            "  (frozen signifies it didn't complete or fail, and is not a monthly download job)",
+            "Ability to filter by an open or closed date range, job status, and if it is a monthly record",
+            "Depending on environment settings, this will either re-queue the download or process locally",
+        ]
+    )
+
+    def add_arguments(self, parser):  # used by parent class
+        parser.add_argument(
+            "-s",
+            "--start",
+            "--start-date",
+            "--start-datetime",
+            dest="start",
+            type=dateutil.parser.parse,
+            help="Earliest datetime to begin restarting downloads",
+        )
+        parser.add_argument(
+            "-e",
+            "--end",
+            "--end-date",
+            "--end-datetime",
+            dest="end",
+            type=dateutil.parser.parse,
+            help="Most recent datetime to restart downloads",
+        )
+        parser.add_argument(
+            "--status",
+            dest="status",
+            choices=[status for status in JOB_STATUS_DICT.keys() if status not in ("finished")],
+            default="ready",
+            help="Choose specific DownloadJob state to filter and process",
+        )
+        parser.add_argument(
+            "--rate-limit",
+            dest="rate-limit",
+            type=int,
+            default=60,
+            help="Time delay in seconds between restarting jobs",
+        )
+        parser.add_argument(
+            "--allow-monthly",
+            dest="monthly",
+            type=bool,
+            default=False,
+            help="If true: include and restart monthly download jobs",
+        )
+
+    def handle(self, *args, **options):  # used by parent class
+        logger.info("Beginning management command")
+
+        filters = self.parse_arguments_to_queryset_filter(**options)
+        list_of_downloads = self.search_for_downloads(filters)
+        logger.info("Found {} Download jobs to restart".format(len(list_of_downloads)))
```

Comment:
In other code, `len(some_list_of_orm_objects)` causes all of them to be lazily loaded from the db just to get a count. If all you're looking for is the count to log, use `.count()` on the filter.

Reply:
Count is just for logging. The next code line loops through the results